### PR TITLE
Add error boundary

### PR DESCRIPTION
**Pull request by RisingGeek**
I tried switching to a branch that has an additional module and then, switching to the previous branch. It works fine if I clear the `localstorage` because the mentioned key is deleted so it needs to be refreshed. Can there be any other possibility of error? #7

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -54,7 +54,9 @@ const App = () => {
     return contributors;
 
   }
-
+  if (selectedModule && selectedComponent && !modules[selectedModule][selectedComponent])
+    localStorage.clear();
```
Comment (AKAspanion) [inline app/index.js:58]: @RisingGeek You should use dispatch to clear out only relevant localStorage keys.

**Comment by AKAspanion** [comment]
@RisingGeek Issue happens when localStorage returns a non-null key, which is then not found in the modules object.

File: app/index.js (hunk 1 above)
Comment (RisingGeek) [reply app/index.js:58]: Okay, I will think of something else because using `dispatch` in the condition will result in multiple re-renders.

**Comment by ManojSatishkumar** [comment]
Is this issue fixed ?
Ready for merge ?

**Comment by RisingGeek** [comment]
> Is this issue fixed ?
> Ready for merge ?

Not yet. Using localStorage fixes the issue but @AKAspanion suggested to use dispatch

**Comment by RisingGeek** [comment]
Issue is now fixed without dispatching actions

File: app/index.js [hunk 2]
```diff
@@ -60,9 +60,11 @@ const App = () => {
...
       {selectedVariant && <LayoutToolbar />}
       {selectedModule && <ComponentsList components={modules[selectedModule].components} />}
-      {selectedComponent && <VariantsList selectedComponent={modules[selectedModule].components[selectedComponent]} />}
-      {selectedVariant && <LayoutArea
-        Variant={modules[selectedModule].components[selectedComponent].variants[selectedVariant]} />}
+      {selectedComponent && modules[selectedModule].components[selectedComponent] &&
+        <VariantsList selectedComponent={modules[selectedModule].components[selectedComponent]} />}
+      {selectedVariant && modules[selectedModule].components[selectedComponent] &&
+        modules[selectedModule].components[selectedComponent].variants[selectedVariant] &&
+        <LayoutArea Variant={modules[selectedModule].components[selectedComponent].variants[selectedVariant]} />}
```
Comment (AKAspanion) [inline app/index.js:67]: image

Giving a default value to Variant prop won't break the component when it tries to access render prop from an undefined Variant. This change can work?

https://github.com/ManojSatishkumar/msk-components/blob/master/app/components/layout-area.js#L26-L44

Comment (RisingGeek) [reply app/index.js:67]: No, if you add a new component add then delete it, `modules[selectedModule].components[selectedComponent]` will be `null` and if you add a new variant and then delete it `modules[selectedModule].components[selectedComponent].variants[selectedVariants]` will be `null`. Error comes out in this file: 
https://github.com/ManojSatishkumar/msk-components/blob/master/app/index.js

Comment (AKAspanion) [reply app/index.js:67]: Yes, saw the changes. That fixes it.

This is just for an optional extra step of error handling on top. 😛
Adding a default value to Variant prop will handle those null/undefined cases.

Comment (RisingGeek) [reply app/index.js:67]: Yeah you could add it optionally. But when `Variant` is an empty object, `Variant.contributors` would be `undefined` and we are mapping over that.

Comment (AKAspanion) [reply app/index.js:67]: All right.

**Comment by RisingGeek** [comment]
@ManojSatishkumar @WebDevRaj @Chiraglr @venkateshpachigulla Please review, this bug is causing problems to many people.

**Comment by ManojSatishkumar** [comment]
You can automatically merge after 3 approvals

**Comment by RisingGeek** [comment]
> You can automatically merge after 3 approvals

okay

**Pull request merged**